packages/utils/src/conversions.ts

```diff
@@ -0,0 +1,202 @@
+/**
+ * @module Conversions
+ * This module provides a collection of utility functions for converting
+ * between different numerical formats, particularly focusing on
+ * conversions involving bigints, hexadecimals and buffers.
+ * The module is structured with clear function naming to indicate
+ * the conversion direction (e.g., `bigIntToHexadecimal` for BigInt
+ * to hexadecimal, `bufferToBigint` for buffer to bigint) and employs
+ * type checks to ensure the correct handling of various input types.
+ * It also includes variations for both big-endian (`be`) and little-endian
+ * (`le`) conversions. It is important to note that when there is no prefix,
+ * the order of bytes is always big-endian.
+ */
+
+import { BigNumberish } from "./types"
+import { isHexadecimal, isStringifiedBigint } from "./number-checks"
+
+/**
+ * Converts a bigint to a hexadecimal string ensuring even length.
+ * @param n The bigint value to convert.
+ * @returns The hexadecimal representation of the bigint.
+ */
+export function beBigIntToHexadecimal(n: bigint): string {
+    let hex = n.toString(16)
+
+    // Ensure even length.
+    if (hex.length % 2 !== 0) {
+        hex = `0${hex}`
+    }
+
+    return hex
+}
+
+/**
+ * Converts a bigint to a hexadecimal string. Alias for beBigIntToHexadecimal.
+ * @param n The bigint value to convert.
+ * @returns The hexadecimal representation of the bigint.
+ */
+export function bigIntToHexadecimal(n: bigint): string {
+    return beBigIntToHexadecimal(n)
+}
+
+/**
+ * Converts a hexadecimal string to a bigint, ensuring the hex string starts with '0x'.
```

Comment:
"ensuring the hex string starts with 0x" isn't behavior which is visible to the caller, so it doesn't seem like the right thing to call out in documentation.

Alternative suggestions: "Coverts a hexadecimal string to a bigint.  The input is interpreted as hexadecimal with or without a `0x` prefix."

Reply:
Right, definitely clearer! Thanks